3rd-party/class.jetpack-amp-support.php

```diff
@@ -409,6 +415,14 @@ public static function amp_enqueue_sharing_css() {
 		}
 	}
 
+	/**
+	 * For the AMP Reader mode template, include styles that we need.
+	 */
+	public static function amp_reader_sharing_css() {
+		echo file_get_contents( plugin_dir_path( dirname( __FILE__ ) ) . '_inc/social-logos/social-logos.css' ); // phpcs:ignore
```

Comment:
@jeherve I see this method of echoing CSS from a file for Reader mode is already used for the Related Posts module. I couldn't think of a better way to do it, or a way to escape CSS, so I added the phpcs:ignore rule.

Reply:
This is a bit problematic, not because of the echoing, but because those 2 files (and their minified versions, which we may want to prefer here) don't live in that same location in the WordPress.com environment.

We could have a check for WordPress.com in this function and echo different files depending on the environment. Unless there is a way we can rely on the styles' enqueue names instead and inject them to the Reader mode's page that way?

Reply:
@jeherve If we went the route of enqueued assets, we'd need to do something like a `wp_remote_get` to fetch the styles.

Quick question, would using `JETPACK__PLUGIN_FILE` instead of `dirname( __FILE__ )` solve this problem?

Reply:
`JETPACK__PLUGIN_FILE` won't do the trick here I'm afraid, as the file's relative locations are different on the 2 platforms.

I've just created #15762 to help a bit, alongside an existing constant we have for the Sharing feature, `WP_SHARING_PLUGIN_DIR`. Once that's merged, we should be able to do something like this:

```diff
diff --git a/3rd-party/class.jetpack-amp-support.php b/3rd-party/class.jetpack-amp-support.php
index a9de75e86..9de0c9ac4 100644
--- a/3rd-party/class.jetpack-amp-support.php
+++ b/3rd-party/class.jetpack-amp-support.php
@@ -419,8 +419,20 @@ class Jetpack_AMP_Support {
 	 * For the AMP Reader mode template, include styles that we need.
 	 */
 	public static function amp_reader_sharing_css() {
-		echo file_get_contents( plugin_dir_path( dirname( __FILE__ ) ) . '_inc/social-logos/social-logos.css' ); // phpcs:ignore
-		echo file_get_contents( plugin_dir_path( dirname( __FILE__ ) ) . 'modules/sharedaddy/amp-sharing.css' ); // phpcs:ignore
+		/*
+		 * We'll need to output the full contents of the 2 files
+		 * in the head on AMP views. We can't rely on regular enqueues here.
+		 *
+		 * phpcs:disable WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		 * phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped
+		 */
+		echo file_get_contents( JETPACK_SOCIAL_LOGOS_DIR . '_inc/social-logos/social-logos.css' );
+		echo file_get_contents( WP_SHARING_PLUGIN_DIR . 'modules/sharedaddy/amp-sharing.css' );
+
+		/*
+		 * phpcs:enable WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		 * phpcs:enable WordPress.Security.EscapeOutput.OutputNotEscaped
+		 */
 	}
 
 	/**

```

Reply:
I've now merged the other PR, you should be good to go with this!